File: cs214/projects/proj10/ruby/Ostrich.rb
# Ostrich.rb | Defines the Ostrich class which inherits attributes and methods
#   from the WalkingBird superclass.
#
# Begun by: Dr. Adams, for CS 214 at Calvin College.
# Completed by: Advait Scaria
# Date: 04/27/2019
####################################################

require './Bird.rb'

class Ostrich < WalkingBird

  ####################################################
  # A Ostrich's Call
  # Return: a Ostrich sound ("Huh-huh-huh-huuuuh").
  ####################################################
  def call
    "Snork"
  end

end

File: cs214/projects/proj10/java/Ostrich.java
/* Ostrich.java provides a Ostrich subclass of WalkingBird.
 *
 * Begun by: Prof. Adams, for CS 214 at Calvin College.
 * Completed by: Advait Scaria
 * Date: 04/27/2019
 ******************************************************/

public class Ostrich extends WalkingBird {

 /* default constructor
  * PostCond: invokes superclass Bird's 
  *     	  default constructor to set
  *			  myName = "".
  */ 
  public Ostrich() {
      super();
  }

 /* explicit constructor
  * Receive: name, a String 
  * PostCond: myName == name.
  */
  public Ostrich(String name)
  {     
      super(name);
  }

 /* A Ostrich's Call
  * Return: an ostrich-call ("Snork").
  */
  public String call()
  {
     return "Snork";
  }

} 

